dnf/cli/commands/repoquery.py

```diff
@@ -393,16 +414,27 @@ def run(self):
             else:
                 q = q.filter(file__glob=self.opts.whatprovides)
         if self.opts.alldeps or self.opts.exactdeps:
-            if not self.opts.whatrequires:
+            if not (self.opts.whatrequires or self.opts.whatdepends):
                 raise dnf.exceptions.Error(
                     _("argument {} requires --whatrequires option".format(
                         '--alldeps' if self.opts.alldeps else '--exactdeps')))
```

Comment:
error message should mention not only --whatrequires, but also --whatdepends option. Also in documentation doc/command_ref.rst